stixcore/products/product.py

```diff
@@ -480,40 +481,85 @@ def __add__(self, other):
         if not isinstance(other, type(self)):
             raise TypeError(f'Products must of same type not {type(self)} and {type(other)}')
 
-        # TODO reindex and update data control_index
+        # make a deep copy of the data and control
         other_control = other.control[:]
         other_data = other.data[:]
-        other_control['index'] = other.control['index'] + self.control['index'].max() + 1
-        control = vstack((self.control, other_control))
-        # control = unique(control, keys=['scet_coarse', 'scet_fine'])
-        # control = control.group_by(['scet_coarse', 'scet_fine'])
+        self_control = self.control[:]
+        self_data = self.data[:]
 
-        other_data['control_index'] = other.data['control_index'] + self.control['index'].max() + 1
+        # keep old index in a unique way
+        other_control['old_index'] = [f"o{i}" for i in other_control['index']]
+        self_control['old_index'] = [f"s{i}" for i in self_control['index']]
+        other_data['old_index'] = [f"o{i}" for i in other_data['control_index']]
+        self_data['old_index'] = [f"s{i}" for i in self_data['control_index']]
 
-        logger.debug('len self: %d, len other %d', len(self.data), len(other_data))
+        control = vstack((self_control, other_control))
 
-        data = vstack((self.data, other_data))
+        logger.debug('len self: %d, len other %d', len(self_data), len(other_data))
+
+        data = (vstack((self_data, other_data))
+                if self.scet_timerange.start <= other.scet_timerange.start
+                else vstack((other_data, self_data)))
 
         logger.debug('len stacked %d', len(data))
 
         # Not sure where the rounding issue is arising need to investigate
         data['time_float'] = np.around(data['time'].as_float().value, 2)
 
+        # remove dublicate data based on time bin and sort the data
         data = unique(data, keys=['time_float'])
+        data.sort(['time_float'])
 
         logger.debug('len unique %d', len(data))
 
         data.remove_column('time_float')
 
-        unique_control_inds = np.unique(data['control_index'])
-        control = control[np.nonzero(control['index'][:, None] == unique_control_inds)[1]]
+        # update the control index in data to a new unique sequence
+        newids = dict()
+
+        for row in data:
+            oid = row['old_index']
+
+            if oid not in newids:
+                newids[oid] = len(newids)
+            nid = newids[oid]
+            row['control_index'] = nid
+
+        del_rows = list()
+
+        # update the index in control as used in data
+        for idx, row in enumerate(control):
+            oid = row['old_index']
+
+            if oid not in newids:
+                del_rows.append(idx)
+                continue
+            nid = newids[oid]
+            row['index'] = nid
```

Comment:
So I think this can be by just dropping the id not in the data table but we can tackle this later